src/frontend/contexts/AppProviders.tsx

```diff
@@ -52,7 +53,9 @@ export const AppProviders = ({
                       <ActiveProjectProvider>
                         <BottomSheetModalProvider>
                           <PhotoPromiseProvider>
-                            <SecurityProvider>{children}</SecurityProvider>
+                            <SecurityProvider>
+                              <MetricsProvider>{children}</MetricsProvider>
```

Comment:
this should probably live further up in this component tree. would say as long as it's higher up than `ActiveProjectProvider`, that should be sufficient.